src/app/util/ember-compatibility-functions.cpp

```diff
@@ -340,5 +344,121 @@ CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * ap
     return CHIP_NO_ERROR;
 }
 
+namespace {
+template <typename T>
+CHIP_ERROR numericTlvDataToAttributeBuffer(TLV::TLVReader & aReader, uint16_t & dataLen)
+{
+    T value;
+    static_assert(sizeof(value) <= sizeof(attributeData), "Value cannot fit into attribute data");
+    dataLen = sizeof(value);
```

Comment:
Should `dataLen` should be set after `ReturnErrorOnFailure(aReader.Get(value));` ?